pkg/cmd/server.go

```diff
@@ -190,6 +197,7 @@ func init() {
 	ServerCmd.Flags().String("artifact-root", "s3://fasttrackml", "Artifact root")
 	ServerCmd.Flags().String("auth-username", "", "BasicAuth username")
 	ServerCmd.Flags().String("auth-password", "", "BasicAuth password")
+	ServerCmd.Flags().String("cors-allow-origins", "*", "CORS allowed origins")
```

Comment:
should this default to all hosts?

Reply:
good question — here is my current thinking:
* figuring out how the server will actually be accessed is not trivial, especially if it's running behind a k8s ingress
* users will usually just run it locally and without any options
* for users that actually care about security, they will now have the option to set the allowed origins

happy to debate it, as I'm not entirely sure about this choice either!

Reply:
actually I'm very wrong about this!
we should instead default to no CORS headers at all when no origin is specified, so that it only works from the same browser origin, and then probably set `*` as the default in the devcontainer

Reply:
what about other CORS options?

Reply:
I have reduced the whole thing to a new hidden `--dev-mode` flag which enables CORS with default options so that `npm start` can be used to develop the UI and the backend together.